src/core/Val.ml

```diff
@@ -784,54 +784,58 @@ struct
     | `Same _ ->
       args
 
-  and rigid_coe_data dir abs el =
-    let x, tyx = Abs.unleash1 abs in
-    match unleash tyx, unleash el with
-    | Data dlbl, Intro info ->
-      (* Figure 8 of Part IV: https://arxiv.org/abs/1801.01568v3; specialized to non-indexed HITs. *)
-      let desc = Sig.lookup_datatype dlbl in
-      let constr = Desc.lookup_constr info.clbl desc in
-      let r, r' = Dir.unleash dir in
+  (* Figure 8 of Part IV: https://arxiv.org/abs/1801.01568v3; specialized to non-indexed HITs. *)
+  and rigid_coe_data_intro dir abs ~dlbl ~clbl ~const_args ~rec_args ~rs =
+    let x, _ = Abs.unleash1 abs in
```

Comment:
I do not understand this line. Why not simply get a fresh variable?

Reply:
indeed, I think I was momentarily confused! Fixing